worker/firewaller/firewaller.go

```diff
@@ -109,10 +109,14 @@ type Config struct {
 
 	CredentialAPI common.CredentialAPI
 
+	// TODO: (jack-w-shaw) Drop these once we move tests to mocks based
 	// WatchMachineNotify is called when the Firewaller starts watching the
 	// machine with the given tag (manual machines aren't watched). This
 	// should only be used for testing.
 	WatchMachineNotify func(tag names.MachineTag)
+	// FlushModelNotify is called when the Firewaller flushes it's model.
+	// This should only be used for testing
+	FlushModelNotify func()
```

Comment:
NOTE: I plan to investigate patching in our tests instead of this horrible nonsense.